File: HW/HW8/solution.py
class Solution:
    def longestPalindrome(self, s):
        chars = [0] * 128
        for ch in s:
            chars[ord(ch)] += 1

        res = 0
        for ch in chars:
            res += ch // 2 * 2
            if res % 2 == 0 and ch % 2 == 1:
                res += 1

        return res


# Test cases
test1 = Solution()

#Provided test Cases
print(test1.longestPalindrome("abccccdd"))      
print(test1.longestPalindrome("speediskey"))   
#Hidden Test Cases
print(test1.longestPalindrome("AAfeiejgA"))   

File: HW/HW8/solution.java
package HW.HW8;

public class solution {
    public int longestPalindrome(String s) {
        int[] chars = new int[128];
        for (char ch : s.toCharArray()){
            chars[ch]++;
        }

        int res = 0;
        for(Integer ch: chars){
            res += ch / 2 * 2;
            if(res % 2 == 0 && ch % 2 == 1){
                res+=1;
            }
        }

        return res;
    }


    public static void main(String[] args) {
        //Provided test cases
        solution test1 = new solution();
        System.out.println(test1.longestPalindrome(("abccccdd")));
        System.out.println(test1.longestPalindrome(("speediskey")));
        //Hidden Test Cases: Dealing with uppercase and lower case values
        System.out.println(test1.longestPalindrome(("AAfeiejgA"))); //Expected 5

    
    }
}
